scripts/ingest_startgg.py

```diff
@@ -3,190 +3,196 @@
+                    "entrant_2_character_ids": set["entrant2CharacterIds"] if "entrant2CharacterIds" in set else [],
+                })
+    return name, sets, date
+
+def process_tournament(slug: str, playlist_urls: list[str], api_key: str):
+    print(f"Processing {slug}")
     data = []
-    slugs = get_slugs_from_url(startgg_url)
-    if not slugs:
-        print("Could not get slugs")
-        return []
-    tournament_slug, event_slug = slugs
-    client = startgg_gql.get_client()
-    tournament_name, timezone = startgg_gql.get_tournament_name_and_timezone(client, tournament_slug)
-    event_id = startgg_gql.get_event_id(client, event_slug)
-    set_ids = startgg_gql.get_event_set_ids(client, event_id)
-    while len(set_ids):
-        set_id = set_ids[-1]
-        print(f"Processing set {set_id}")
-        try:
-            set_data = startgg_gql.get_set_data(client, set_id)
-            vod_data = get_vod_data(set_data, tournament_name, timezone)
-            if not vod_data:
-                # Try to get video from playlist
-                url = find_matching_youtube_video(set_data, playlist_urls)
-                if url:
-                    set_data["vodUrl"] = url
-                    vod_data = get_vod_data(set_data, tournament_name, timezone)
-                    try:
-                        startgg_gql.set_vod(client, set_id, url)
-                    except Exception as e:
-                        print(f"Failed to assign video: {e}")
-            if vod_data:
-                vod_data["id"] = len(data)
-                data.append(vod_data)
-            set_ids.pop()
-        except TransportServerError as err:
-            if err.code == 429:
-                print("Too many requests; taking a nap")
-                print("(You shouldn't see this if you're limiting reqs properly)")
-                client = startgg_gql.get_client()
-                time.sleep(10)
+    name, sets, date = get_tournament_name_sets_and_timezone(slug)
+
+    set_id_to_video_url = dict[str, str]()
+    current_playlist_urls = playlist_urls
+    unmatched_playlist_urls = []
+    while len(current_playlist_urls):
+        next_playlist_urls: list[str] = []
+        for playlist_url in current_playlist_urls:
+            if playlist_url is None:
+                continue
+            video_title, video_url = playlist_url
+            video_title_safe = normalize(video_title)
+            phase_matching_sets = []
+            matching_sets = []
+            for set in sets:
+                all_gamer_tags_found = True
+                for gamer_tag in set["entrant_1_gamer_tags"] + set["entrant_2_gamer_tags"]:
+                    if normalize(gamer_tag) not in video_title_safe:
+                        all_gamer_tags_found = False
+                        break
+                if all_gamer_tags_found and (normalize(set["full_round_text"]) in video_title_safe):
```

Comment:
Requiring the full round name be present is making this not work for our tournaments, where we do short round titles, e.g. GF for Grand Finals. I initially avoided including rounds at all because of how much variation there can be around video titles in general. Though I don't think it's possible to make this automated and *not* include round titles.

Moving the round text check to the next conditional made things work better for me.